pymc3/math.py

```diff
@@ -136,19 +176,28 @@ def logsumexp(x, axis=None):
 
 def logaddexp(a, b):
     diff = b - a
-    return tt.switch(diff > 0,
-                     b + tt.log1p(tt.exp(-diff)),
-                     a + tt.log1p(tt.exp(diff)))
+    return tt.switch(diff > 0, b + tt.log1p(tt.exp(-diff)), a + tt.log1p(tt.exp(diff)))
 
 
 def logdiffexp(a, b):
     """log(exp(a) - exp(b))"""
     return a + log1mexp(a - b)
 
 
+def logdiffexp_numpy(a, b):
+    """log(exp(a) - exp(b))"""
+    return a + log1mexp_numpy(a - b)
+
+
 def invlogit(x, eps=sys.float_info.epsilon):
     """The inverse of the logit function, 1 / (1 + exp(-x))."""
-    return (1. - 2. * eps) / (1. + tt.exp(-x)) + eps
+    return (1.0 - 2.0 * eps) / (1.0 + tt.exp(-x)) + eps
+
+
+def logbern(log_p):
+    if np.isnan(log_p):
+        raise FloatingPointError("log_p can't be nan.")
```

Comment:
wow, who knew `FloatingPointError` is a builtin?

Reply:
If only pm.exceptions.ShapeError was a builtin too...